warehouse/packaging/models.py

```diff
@@ -221,6 +225,10 @@ def __acl__(self):
             (Allow, "group:moderators", "moderator"),
         ]
 
+        # The project itself has an identity (for OIDC-minted tokens),
+        # and that identity has the ability to upload releases.
+        acls.append((Allow, f"project:{self.id}", ["upload"]))
```

Comment:
Flagging: This is how we give each `Project` identity permission to upload to itself. I'm doing this the same way we do `User` identities, but just flagging it in case I'm missing anything.